provisioning/transport/amqp/src/amqp.ts

```diff
@@ -125,11 +126,23 @@ export class Amqp extends EventEmitter implements X509ProvisioningTransport, Tpm
           _onEnter: (request, callback) => {
             /*Codes_SRS_NODE_PROVISIONING_AMQP_16_002: [The `registrationRequest` method shall connect the AMQP client with the certificate and key given in the `auth` parameter of the previously called `setAuthentication` method.]*/
             /*Codes_SRS_NODE_PROVISIONING_AMQP_16_012: [The `queryOperationStatus` method shall connect the AMQP client with the certificate and key given in the `auth` parameter of the previously called `setAuthentication` method. **]**]*/
-            const config: AmqpBaseTransportConfig = {
+            let config: AmqpBaseTransportConfig = {
               uri: this._getConnectionUri(request),
               sslOptions: this._x509Auth,
               userAgentString: ProvisioningDeviceConstants.userAgent
             };
+            if (this._sasToken) {
```

Comment:
>if (this._sasToken)  [](start = 12, length = 20)

ok, just realizing this now. that should *not* be in the x509 path.

Reply:
I was in a quandary here.  The fact is the path is exactly the same between the two, with the addition of setting the sas token.  Basically I could have changed this states name or created a whole new state with a symmetric name and repeating this code.  The repeating code was not good.  Plus the decision of which state seemed quite specious.  So then it was like what state name to change it too.  Maybe just changing it to connectiongX509OrSymmetric would be ok.  However, I wasn't particularly happy with the code here anyway.  I certainly don't like the odd interaction with the connectingTpm.  This seemed funky.  At that point, given the choice of doing a bit of re-factoring of the state machine (for little functional benefit), making a name change for minor improvement in readability, or simply go with what I had, I decided to go with what I had.  So, what I'm going to do now is change the state name.

---
In reply to: [231753929](https://github.com/Azure/azure-iot-sdk-node/pull/418#discussion_r231753929) [](ancestors = 231753929)

Reply:
I'm ok with renaming the state. the whole point is that it should be easy for whoever reads this code next to know why it's written that way and what it's doing.

---
In reply to: [232002313](https://github.com/Azure/azure-iot-sdk-node/pull/418#discussion_r232002313) [](ancestors = 232002313,231753929)